src/main/scala/com/amazon/deequ/checks/Check.scala

```diff
@@ -1355,4 +1399,117 @@ object Check {
 
     detectedAnomalies.anomalies.isEmpty
   }
+
+
+  /**
+   * Common assertion function checking if the value can be considered as normal (that no
+   * anomalies were detected), given the anomaly detection strategy with extended results
+   * and details on how to retrieve the history.
+   * This assertion function returns an AnomalyDetectionAssertionResult which contains
+   * anomaly detection extended results.
+   *
+   * @param metricsRepository        A metrics repository to get the previous results.
+   * @param anomalyDetectionStrategyWithExtendedResults The anomaly detection strategy with extended results.
+   * @param analyzer                 The analyzer for the metric to run anomaly detection on.
+   * @param withTagValues            Can contain a Map with tag names and the corresponding values
+   *                                 to filter for.
+   * @param beforeDate               The maximum dateTime of previous AnalysisResults to use for
+   *                                 the Anomaly Detection.
+   * @param afterDate                The minimum dateTime of previous AnalysisResults to use for
+   *                                 the Anomaly Detection.
+   * @param currentMetricValue       current metric value.
+   * @return The AnomalyDetectionAssertionResult with the boolean if the newest data point is anomalous
+   *         along with the AnomalyDetectionExtendedResult object which contains the
+   *         anomaly detection extended result details.
+   */
+  private[deequ] def isNewestPointNonAnomalousWithExtendedResults[S <: State[S]](
```

Comment:
Should we change the name of this method? The `is<>` implies returning a boolean. Since we are returning a case class, perhaps we can update the method name to imply that.